src/QsCompiler/DataStructures/ReservedKeywords.fs

```diff
@@ -267,6 +267,7 @@ module AssemblyConstants =
     let QCIProcessor = "QCIProcessor"
     let QuantinuumProcessor = "QuantinuumProcessor"
     let RigettiProcessor = "RigettiProcessor"
+    let MicrosoftQuantum = "MicrosoftQuantum"
```

Comment:
I don't think there's a need to introduce a new processor architecture. `MicrosoftQuantum` as a target might encompass multiple processor architectures but for now there's no need to introduce a new one.

Reply:
Renamed all instances of MicrosoftSimulator to MicrosoftQuantum so that there is no leakage of old architecture with new provider